js/temperature_humidity_sensor.js

```diff
@@ -0,0 +1,53 @@
+export class TemperatureHumiditySensor {
+    constructor(data) {
+        this.data = data
+    }
+
+    data() {
+        return this.data
+    }
+
+    get inside() {
+        const inside_data = this.data.find(r => r.name == "Music room")
+        return inside_data ? new Location(inside_data) : new EmptyLocation()
+    }
+
+    get outside() {
+        const outside_data = this.data.find(r => r.name == "Front porch")
+        return outside_data ? new Location(outside_data) : new EmptyLocation()
+    }
+}
+
+export class Location {
```

Comment:
I wonder if we should call this something like `LocationWeatherMeasurement`?

Reply:
I think it's OK to have this be less specific because it's in the file temperature_humidity_sensor.js. The file acts like a module, so a Location object in there has that context. If there's a collision we can use `import {Location as LocationWeatherMeasurement} from ... `